ui/components/app/currency-input/currency-input.js

```diff
@@ -148,48 +161,39 @@ export default function CurrencyInput({
       return null;
     }
 
-    if (shouldUseFiat) {
-      // Display ETH
-      currency = preferredCurrency || EtherDenomination.ETH;
-      numberOfDecimals = 8;
+    if (isTokenPrimary) {
+      // Display fiat; `displayValue` bypasses calculations
+      displayValue = formatCurrency(
+        new Numeric(fiatDecimalValue, 10).toString(),
+        secondaryCurrency,
+      );
     } else {
-      // Display Fiat
-      currency = isOriginalNativeSymbol ? secondaryCurrency : null;
-      numberOfDecimals = 2;
+      // Display token
+      suffix = primarySuffix;
+      displayValue = new Numeric(tokenDecimalValue, 10).toString();
     }
 
     return (
       <CurrencyDisplay
+        // hides the fiat suffix
+        hideLabel={isTokenPrimary || isLongSymbol}
+        suffix={suffix}
         className="currency-input__conversion-component"
-        currency={currency}
-        value={newHexValue}
-        numberOfDecimals={numberOfDecimals}
+        displayValue={displayValue}
       />
     );
   };
 
   return (
     <UnitInput
-      {...{
-        hexValue,
-        preferredCurrency,
-        secondaryCurrency,
-        hideSecondary,
-        featureSecondary,
-        conversionRate,
-        onChange,
-        onPreferenceToggle,
-      }}
+      hideSuffix={isTokenPrimary && isLongSymbol}
       dataTestId="currency-input"
-      suffix={
-        shouldUseFiat && isOriginalNativeSymbol
-          ? secondarySuffix
-          : primarySuffix
-      }
-      onChange={handleChange}
-      value={initialDecimalValue}
+      suffix={isTokenPrimary ? primarySuffix : secondarySuffix}
+      onChange={handleChange ?? onChange}
```

Comment:
Is this fallback needed? Isn't handleChange always defined?

Reply:
No I'm tweaking. Good catch